### PR TITLE
Reenable native tools bootstrapping

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/dotnet/arcade/issues/2533

After reenabling the bootstrapping I hit serveral issues which I also fixed as part of this PR:
- global.json path was wrong in some situations
- `native_tools_list` variable in the shell script contained zero-width whitespace characters which overwrote the first few characters in a path.
- Wrong condition in the shell script when using the clean option
- Inconsistent console outputs between Unix and Windows

Changes:
- When using with CI the bin directory now defaults to `$(RepoRoot)/.tools/native/`

Tested on both Ubuntu 18.04 and Windows 10 1903.

cc @ericstj

### CODE REVIEW

**Comment by ViktorHofer** [comment]
PTAL @tmat @cchosta

**Comment by ViktorHofer** [comment]
PTAL @alexperovich @tmat @chcosta

File: eng/common/tools.ps1 [hunk 1]
```diff
@@ -391,6 +391,17 @@ function GetSdkTaskProject([string]$taskName) {
   return Join-Path (Split-Path (InitializeToolset) -Parent) "SdkTasks\$taskName.proj"
 }
 
+function InitializeNativeTools() {
+  $nativeTools = $GlobalJson | Select-Object -Expand "native-tools" -ErrorAction SilentlyContinue
+  if ($nativeTools) {
```
Comment (tmat) [inline eng/common/tools.ps1]: Simpler: `if (Get-Member -InputObject $GlobalJson -Name "native-tools")`

File: eng/common/tools.ps1 [hunk 2]
```diff
@@ -391,6 +391,17 @@ function GetSdkTaskProject([string]$taskName) {
...
+    $nativeArgs=""
+    if ($ci) {
+      $nativeArgs = "-InstallDirectory " + (Join-Path $ToolsDir "native")
```
Comment (tmat) [inline eng/common/tools.ps1]: I think we could install the tools directly to `$ToolsDir`, without the extra subdir.

File: eng/common/init-tools-native.sh [hunk 3]
```diff
@@ -116,8 +117,8 @@ else
       installer_command+=" --clean"
     fi
 
-    echo "Installing $tool version $tool_version"
-    echo "Executing '$installer_command'"
+    # echo "Installing $tool version $tool_version"
```
Comment (tmat) [inline eng/common/init-tools-native.sh]: Remove the lines.

File: eng/common/init-tools-native.ps1 [hunk 4]
```diff
@@ -41,9 +41,13 @@ Param (
   [switch] $Force = $False,
   [int] $DownloadRetries = 5,
   [int] $RetryWaitTimeInSeconds = 30,
-  [string] $GlobalJsonFile = "$PSScriptRoot\..\..\global.json"
+  [string] $GlobalJsonFile
 )
 
+if (!$GlobalJsonFile) {
+  $GlobalJsonFile = Join-Path (get-item $PSScriptRoot).parent.parent.FullName "global.json"
```
Comment (tmat) [inline eng/common/init-tools-native.ps1]: Nit: Capitalize `Get-Item`, `Parent`

**Pull request merged**